R/data_extract_module.R

```diff
@@ -332,18 +344,58 @@ check_data_extract_spec_react <- function(datasets, data_extract) {
 #' runApp(app)
 #' }
 #'
-data_extract_srv <- function(id, datasets, data_extract_spec) {
-  checkmate::assert_class(datasets, "FilteredData")
+data_extract_srv <- function(id, datasets, data_extract_spec, ...) {
+  checkmate::assert_multi_class(datasets, c("FilteredData", "list"))
   stopifnot(
     "`data_extract_spec` argument should be of \"data_extract_spec\" class or a list with all elements of such class" =
       inherits(data_extract_spec, "data_extract_spec") ||
         (is.list(data_extract_spec) && all(vapply(data_extract_spec, inherits, logical(1), "data_extract_spec")))
   )
+  UseMethod("data_extract_srv", datasets)
+}
+
+#' @rdname data_extract_srv
+#' @export
+data_extract_srv.FilteredData <- function(id, datasets, data_extract_spec) {
+  checkmate::assert_class(datasets, "FilteredData")
   moduleServer(
     id,
     function(input, output, session) {
       logger::log_trace(
-        "data_extract_srv initialized with datasets: { paste(datasets$datanames(), collapse = ', ') }."
+        "data_extract_srv.FilteredData initialized with datasets: { paste(datasets$datanames(), collapse = ', ') }."
+      )
+
+      data_list <- sapply(X = datasets$datanames(), simplify = FALSE, FUN = function(x) {
+        isolate(datasets$get_data(dataname = x, filtered = TRUE))
+      })
```

Comment:
I think it should be a list of `reactive(<object>)`, otherwise module will not react to the changes in data.
```suggestion
      data_list <- sapply(X = datasets$datanames(), simplify = FALSE, FUN = function(x) {
        reactive(datasets$get_data(dataname = x, filtered = TRUE))
      })
```